launcher/LaunchController.cpp

```diff
@@ -283,6 +282,34 @@ void LaunchController::login()
     emitFailed(tr("Failed to launch."));
 }
 
+bool LaunchController::reauthenticateCurrentAccount()
+{
+    auto button =
+        QMessageBox::warning(m_parentWidget, tr("Account refresh failed"),
+                             tr("The account has expired and needs to be reauthenticated. Do you want to reauthenticate this account?"),
+                             QMessageBox::StandardButton::Yes | QMessageBox::StandardButton::No, QMessageBox::StandardButton::Yes);
+    if (button == QMessageBox::StandardButton::Yes) {
+        auto accounts = APPLICATION->accounts();
+        bool isDefault = accounts->defaultAccount() == m_accountToUse;
+        accounts->removeAccount(accounts->index(accounts->findAccountByProfileId(m_accountToUse->profileId())));
+        if (m_accountToUse->accountType() == AccountType::MSA) {
+            auto newAccount =
+                MSALoginDialog::newAccount(m_parentWidget, tr("Please enter your Mojang account email and password to add your account."));
+            accounts->addAccount(newAccount);
+            if (isDefault) {
+                accounts->setDefaultAccount(newAccount);
+            }
+            m_accountToUse = nullptr;
+            decideAccount();
+            return true;
```

Comment:
does this need to be updated?

Reply:
In what way? Do you mean to bring it up to date with the latest develop?

Reply:
yes, the `MSALoginDialog::newAccount` call looks wrong for some reason.
also should we be changing the translation string considering mojang accounts don't really exist anymore?

Reply:
That message is quickly overridden by another message (the subtask status message).
And no I did not have any issues with updating to latest develop.